roles/translator/src/upstream_sv2/upstream.rs

```diff
@@ -717,25 +724,29 @@ impl ParseUpstreamMiningMessages<Downstream, NullDownstreamMiningSelector, NoRou
     /// Handles the SV2 `OpenExtendedMiningChannelError` message (TODO).
     fn handle_open_mining_channel_error(
         &mut self,
-        _: roles_logic_sv2::mining_sv2::OpenMiningChannelError,
+        m: roles_logic_sv2::mining_sv2::OpenMiningChannelError,
     ) -> Result<roles_logic_sv2::handlers::mining::SendTo<Downstream>, RolesLogicError> {
-        todo!()
+        Ok(SendTo::None(Some(Mining::OpenMiningChannelError(
+            m.as_static(),
+        ))))
     }
 
     /// Handles the SV2 `UpdateChannelError` message (TODO).
     fn handle_update_channel_error(
         &mut self,
-        _m: roles_logic_sv2::mining_sv2::UpdateChannelError,
+        m: roles_logic_sv2::mining_sv2::UpdateChannelError,
     ) -> Result<roles_logic_sv2::handlers::mining::SendTo<Downstream>, RolesLogicError> {
-        todo!()
+        Ok(SendTo::None(Some(Mining::UpdateChannelError(
+            m.as_static(),
+        ))))
     }
 
     /// Handles the SV2 `CloseChannel` message (TODO).
     fn handle_close_channel(
         &mut self,
-        _m: roles_logic_sv2::mining_sv2::CloseChannel,
+        m: roles_logic_sv2::mining_sv2::CloseChannel,
     ) -> Result<roles_logic_sv2::handlers::mining::SendTo<Downstream>, RolesLogicError> {
-        todo!()
+        Ok(SendTo::None(Some(Mining::CloseChannel(m.as_static()))))
```

Comment:
`SendTo::None(Some(Mining::CloseChannel)` is not matched in match next_message_to_send this lead to a panic

Reply:
This  was the one missing, I reviewed the code on vim then jumped here and wrote that comment for the wrong handler sorry